lxd/apparmor.go

```diff
@@ -63,17 +79,35 @@ profile "%s" flags=(attach_disconnected,mediate_deleted) {
     change_profile -> "%s",
 }`
 
-func AAProfileFull(c container) string {
-	lxddir := shared.VarPath("")
-	if len(c.Name())+len(lxddir)+7 >= 253 {
+func mkApparmorName(name string) string {
+	if len(name)+7 >= 253 {
 		hash := sha256.New()
-		io.WriteString(hash, lxddir)
-		lxddir = fmt.Sprintf("%x", hash.Sum(nil))
+		io.WriteString(hash, name)
+		return fmt.Sprintf("%x", hash.Sum(nil))
 	}
 
+	return name
+}
+
+func AANamespace(c container) string {
+	/* / is not allowed in apparmor namespace names; let's also trim the
+	 * leading / so it doesn't look like "-var-lib-lxd"
+	 */
+	lxddir := strings.Replace(shared.VarPath("")[1:], "/", "-", -1)
```

Comment:
On Fri, Jun 24, 2016 at 08:33:56AM -0700, Sean Christopherson wrote:
> >  	}
> >  
> > +	return name
> > +}
> > +
> > +func AANamespace(c container) string {
> > +	/* / is not allowed in apparmor namespace names; let's also trim the
> > +	 * leading / so it doesn't look like "-var-lib-lxd"
> > +	 */
> > +	lxddir := strings.Replace(shared.VarPath("")[1:], "/", "-", -1)
> 
> VarPath isn't guaranteed to start with a `/`, probably should use `strings.Trim` instead of unconditionally slicing off the first character.  A full trim will also remove any trailing slashes.

Thanks, I just added a call to Trim.

FWIW, I don't think we're planning to land this in its current form
yet; there's still some stuff we need to talk about with the apparmor
guys to work out exactly how it should work.

> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/lxc/lxd/pull/2151/files/2f93569e660ded55d554a4a4e6a8065d94835b42#r68416301